### PR TITLE
Text Area Max Length

### LINKED ISSUE #705: [Feature] Max Length for Text Area
### What is the feature that you would like to propose?
I am requesting that we expose the `maxlength` attribute for the `text area`. There is already an issue for exposing the maxlength attribute on inputs, but since the text-area is a different component we will also need to expose it there as well

### Describe the solution you'd like
On a `go-text-area`, be able to pass a value `maxlength` and have it apply to the `text-area` element inside the `go-text-area`.

### PR DESCRIPTION
**Pull request by StevenUlmer**
## PR Checklist
Please check if your PR fulfills the following requirements:
- [x] The commit message follows our [guidelines](https://github.com/mobi/goponents/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added
- [x] Docs have been added or updated

## PR Type
What kind of change does this PR introduce?
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Style Update (CSS)
- [ ] Other... Please describe:

## What is the current behavior?
Resolves #705 

## What is the new behavior?
There is the ability to pass in a maxlength on the text area

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/go-style-guide/src/app/features/ui-kit/components/form-docs/components/text-area-docs/text-area-docs.component.html [hunk 1]
```diff
@@ -345,4 +345,42 @@ <h4 class="go-heading-4 go-heading--underlined">Code</h4>
...
+  <go-card id="form-textarea-max-length" class="go-column go-column--100">
+    <ng-container go-card-header>
+      <h2 class="go-heading-2 go-heading--no-wrap">Component Max Length</h2>
+      <go-copy cardId="form-textarea-max-length" goCopyCardLink></go-copy>
+    </ng-container>
+    <div class="go-container" go-card-content>
+
+      <div class="go-column go-column--100">
+        <p class="go-body-copy go-body-copy--no-margin">
+          The text area component's max legnth can be set via
```
Comment (jaredami) [inline projects/go-style-guide/src/app/features/ui-kit/components/form-docs/components/text-area-docs/text-area-docs.component.html:358]: Should "legnth" be "length"?

**Review by jaredami — CHANGES_REQUESTED** [review]
Just noticed a typo but looks good otherwise 👍

**Comment by grahamhency** [comment]
@StevenUlmer, what if we also add on the `minlength` attribute to this? Probably not going to be used very often, but we're doing it in this PR as well #699

**Review by grahamhency — DISMISSED** [review]
Looks good, thanks!

File: projects/go-lib/src/lib/components/go-text-area/go-text-area.component.html [hunk 2]
```diff
@@ -15,6 +15,8 @@
     [placeholder]="placeholder"
     [rows]="rows"
     [formControl]="control"
+    [attr.maxlength]="maxlength || null"
+    [attr.minlength]="minlength || null"
```
Comment (jaredami) [inline projects/go-lib/src/lib/components/go-text-area/go-text-area.component.html:19]: One question I have about this is whether we should be using `attr.` for these. In the [Angular docs](https://angular.io/guide/binding-syntax) it says: "It is recommended that you set an element property with a property binding whenever possible."

Also [here](https://angular.io/guide/binding-syntax) it says "Attributes initialize DOM properties and then they are done. Property values can change; attribute values can't." I'm wondering if that means that when you use `attr.` it makes the binding static (i.e. it won't update if the value is changed)?

These two things make me think maybe we should just property binding here, unless the `text-area`s don't have these properties.

**Review by jaredami — CHANGES_REQUESTED** [review]
Couple of things in addition to my question about attribute vs property binding:

1. Should we add default values and add some logic to ensure that the `minlength` can't be set to greater than the `maxlength`, like what was done [here](https://github.com/mobi/goponents/pull/699#event-3975155105)?
2. Should we *technically* use brackets when setting the binding values since the inputs are of type `number` rather than `string`? Maybe not worth caring about...

**Review by grahamhency — APPROVED** [review]

**Pull request merged**